src/main/ipv-stub/service/dynamodb-form-response-service.ts

```diff
@@ -1,3 +1,72 @@
 import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { DynamoDBDocument } from "@aws-sdk/lib-dynamodb";
+
+import { UserIdentity } from "../types";
 
 const client = new DynamoDBClient({});
+const dynamo = DynamoDBDocument.from(client);
+
+const tableName = `${process.env.ENVIRONMENT}-UserIdentity`;
+
+export const getUserIdentityWithAuthCode = async (
+  authCode: string
+): Promise<UserIdentity> => {
+  const userIdentity = await dynamo.get({
+    TableName: tableName,
+    Key: { UserIdentityId: authCode },
+  });
+  return userIdentity as unknown as UserIdentity;
```

Comment:
What's this doing? seems a bit odd

Reply:
Typecasting that when we get the object back from the database we can use it as that object type. Not sure if there is a better way to typecast.

Reply:
The dynamo.get method returns a  `GetCommandOutput` which looks something like this 
```
{
$metadata: ResponseMetadata,
Item?: Record<string, AttributeValue>,
ConsumedCapacity?: ConsumedCapacity,
}

```
so you'll want to pull out the field `userIdentity.Item` to return the actual item. This is why the typecast is a bit more complicated here, you should be able to simplify this to `return userIdentity.Item as UserIdentity`

Reply:
Ah nice spot @Ryan-Andrews99 - thanks